File: src/Model/Expression/VariableExpression.java
package Model.Expression;

import DataTypes.MyIDictionary;
import Exception.ExpressionEvaluationException;
import Exception.UndeclaredVariableException;
import Exception.InvalidKeyMyDictionaryException;

/**
 * Created by vladc on 12.10.2016.
 */
public class VariableExpression extends Expression {
    private String id;

    public VariableExpression(String id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return id;
    }

    public int evaluate(MyIDictionary<String, Integer> symTab) throws ExpressionEvaluationException {
        try {
            return symTab.get(id);
        }
        catch(InvalidKeyMyDictionaryException e) {
            throw new UndeclaredVariableException("Exception. Undeclared variable: " + id + ".");
        }
    }
}

File: Toy-Language-Interpreter/Model/Expression/VariableExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.DataTypes;
using Toy_Language_Interpreter.MyException;

namespace Toy_Language_Interpreter.Model.Expression
{
    class VariableExpression : AExpression
    {
        private string id;

        public VariableExpression(string id)
        {
            this.id = id;
        }

        public override int Evaluate(MyIDictionary<string, int> symTab)
        {
            try
            {
                return symTab.Get(id);
            }
            catch (InvalidKeyMyDictionaryException)
            {
                throw new UndeclaredVariableException("Exception: Undeclared variable: " + id + ".");
            }
        }

        public override string ToString()
        {
            return id;
        }
    }
}
